aiida/backends/djsite/db/subtests/migrations.py

```diff
@@ -43,14 +43,22 @@ def setUp(self):
         # Reverse to the original migration
         executor.migrate(self.migrate_from)
 
-        self.setUpBeforeMigration(old_apps)
+        try:
+            self.setUpBeforeMigration(old_apps)
+            # Run the migration to test
+            executor = MigrationExecutor(connection)
+            executor.loader.build_graph()
+            executor.migrate(self.migrate_to)
+
+            self.apps = executor.loader.project_state(self.migrate_to).apps
+        except Exception:
+            # Bring back the DB to the correct state if this setup part fails
+            from ..migrations import LATEST_MIGRATION
```

Comment:
Maybe put this in a function `revert_database_schema` such that you can deduplicate the code in the `tearDown` method